src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -153,8 +166,348 @@ public User updateUser(User user) {
+   * @param game: Game which shall be deleted
+   */
+  public void deleteGameIfEmpty(User user, Game game) {
+    // if game is not active and has no more plays, we can delete it.
+    if(game.getPlays().isEmpty() && game.getGameStatus() != GameStatus.ACTIVE) {
+      user.deletePastGame(game);
+    }
+    userRepository.saveAndFlush(user);
+  }
+
+  /**
+   * Move Active Game to Past Games IF the time is up
+   * @param userId: userId from User
+   */
+  public void updateActiveGameIfNecessary(Long userId) {
+      updateActiveGameIfNecessary(getUserById(userId));
+  }
+
+  /**
+   * Move Active Game to Past Games IF the time is up
+   * @param user: User from whom the active games shall be updated
+   */
+  private void updateActiveGameIfNecessary(User user) {
+    Game activeGame = user.getActiveGame();
+    // case there is no active game, just return
+    if (activeGame == null) return;
+
+    // calculate how old the active game is
+    long diffTime = new Date().getTime() - activeGame.getCreationTime().getTime();
+    // case the game is older than one Day, put it to the past games
+    if(diffTime > Time.ONE_DAY) {
+      // update the user who was a candidate for black cards
+      activeGame.setGameStatus(GameStatus.INACTIVE);
+      user.flushGameToPastGames();
+      // TODO: 12.04.2022 SaveAndFlush GameRepository here? (IDE doesn't complain until now)
+      userRepository.saveAndFlush(user);
+    }
+    // case the active game is not older than 24 hours, just return
+  }
+
+  /**
+   * Get the current black cards of a user: automatically delete old black cards that are older than some time period
+   * @param userId: userId from the user from whom we want to get the current black cards
+   * @return: a list of black cards - can be empty
+   */
+  public List<BlackCard> getCurrentBlackCards(Long userId) {
+    User user = getUserById(userId);
+    // get current black cards
+    UserCards currentCards = user.getUserCards();
+    // return empty list if user has no current black cards (last two conditions are a safety net)
+    if (currentCards == null || currentCards.getBlackCards() == null || currentCards.getBlackCards().isEmpty()) {
+      return Collections.emptyList();
+    }
+    // return current black cards if they're younger than one day, else return empty list
+    long diffTime = new Date().getTime() - currentCards.getDate().getTime();
+    if(diffTime > Time.ONE_DAY) {
+      user.setUserCards(null);
+      userRepository.saveAndFlush(user);
+      return Collections.emptyList();
+    } else {
+      return currentCards.getBlackCards();
+    }
+  }
+
+  /**
+   * Given a List of Black Cards, we want the user to hold these black cards for some time period.
+   * Precondition: user.getCards() is null
+   * @param userId: userId from user to whom the black cards shall be assigned
+   * @param cards: List of black Cards
+   */
+  public void assignBlackCardsToUser(Long userId, List<BlackCard> cards) {
```

Comment:
Why list of black cards and not only one? What am i missing here?

Reply:
Those black cards are not the black cards for the active game(s) for the user, but the 8 black cards that a user can choose from.